presto-main/src/main/java/com/facebook/presto/server/SessionContext.java

```diff
@@ -80,4 +81,16 @@ default List<X509Certificate> getCertificates()
     boolean supportClientTransaction();
 
     Map<SqlFunctionId, SqlInvokedFunction> getSessionFunctions();
+
+    default void setAuthorizedIdentity(
+            Identity identity,
+            Optional<String> selectedUser,
+            Optional<String> reasonForSelection)
+    {
+    }
+
+    default Session appendAuthorizedIdentity(Session session)
```

Comment:
An append method in the session context, which appends to the session object seems out of place to me. We should only have this in the session builder, and sessioncontext should only have getter methods.

Reply:
Then how does the session context know where to get the authorized identity? The session context has been created before we call selectAuthorizedIdentity.

Reply:
I don't think I follow. I think we just need to move this method, the logic remains same.

Reply:
If sessionContext.getAuthorizedIdentity to the id, then sessionContext need to know where to fetch it. Since authorizedIdentity is generated after sessionContext, we need to assign it to sessionContext before sessionContext can give it to session. Right?

Reply:
I have a similar comment, the current code change does not fit with the existing design and it is hacky and will be hard to maintain.